File: leetcode/70.kt
class Solution {
    fun climbStairs(n: Int): Int {
        tailrec fun fib(
            n: Int,
            a: Int,
            b: Int,
        ): Int =
            when (n) {
                0 -> a
                1 -> b
                else -> fib(n - 1, b, a + b)
            }
        return fib(n + 1, 0, 1)
    }
}

File: leetcode/70.c
int climbStairs(int n){
    int fib(int n, int a, int b) {
        if (n == 0) {
            return a;
        } else if ( n == 1 ) {
            return b;
        } else {
            return fib(n-1, b, a + b);
        }
    }
  
    return fib(n+1, 0, 1);
}
